### DISCUSSION TITLE
Separate searchindexes for frontend and control panel?

### DISCUSSION THREAD

**Discussion by KarlBishop** [craftcms/cms#10536] [Ideas] [answered]
We had some performance issues with bloat in our `searchindex` table.

I'm not sure what caused it to get so big but it was at 1.7 million rows ~270MB.

We've stripped it back for now by:
- Unticking "Use this field’s values as search keywords" on all custom fields
- Installing [Oliver Stark's Relax plugin](https://github.com/ostark/craft-relax)
- Clearing the table using `TRUNCATE TABLE my_db_name.searchindex;`
- Rebuilding the table by [resaving entries, users and categories](https://craftcms.com/docs/3.x/console-commands.html#resave) with the `--update-search-index` switch

The table is now down to 78 thousand rows ~15MB and search is much, much faster (if limited).

We plan to re-add custom fields to the `searchindex` after seeing how this performs for a while. But it occurred to me that **most custom fields we used as search keywords were purely to benefit Control Panel users**. The search form on the frontend leveraged only a couple of custom fields that were not intended for use in the CP.

This may be a crazy/stupid idea but I wondered if it would make sense to allow the use of separate `searchindex` tables for the frontend and CP?

**Comment by brandonkelly** [comment, marked as answer]
> But it occurred to me that **most custom fields we used as search keywords were purely to benefit Control Panel users**.

I have a feeling there won’t be as much of a need for this in Craft 4+, thanks to element index filtering (#10075), which is more precise and efficient than search queries.

**Reply by KarlBishop** [reply]
Ahh OK, would this allow an Entries Query on a search results page (like [the one in the docs](https://craftcms.com/knowledge-base/search-form)) to specify a set of fields to filter against?

E.g. something like this?

    {% set searchQuery = craft.app.request.getParam('q') %}
    {% paginate craft.entries()
      .search(searchQuery)
      .filterFields(['title', 'subheadline', 'matrixContentBuilder:redactorField'])
      .section(['news', 'events', 'pressReleases'])
      .orderBy('score')
      .limit(12)
      .with(['heroImage'])
      as entries %}

**Reply by brandonkelly** [reply]
It will allow you to create filters/conditions for specific fields and attributes, using a typical condition builder/smart query UI.
